### PR TITLE
add new time range and grain configs

### PR DESCRIPTION
**Pull request by pjain1**

### CODE REVIEW

**Comment by pjain1** [comment]
@begelundmuller can you have a look ? Thanks

**Comment by pjain1** [comment]
Using https://github.com/senseyeio/duration library to validate ISO duration which is MIT licensed, hope this is fine.

**Review by begelundmuller — CHANGES_REQUESTED** [review]
Just some small comments, otherwise this looks great

File: go.mod [hunk 1]
```diff
@@ -23,6 +23,7 @@ require (
 	github.com/labstack/echo/v4 v4.9.0
 	github.com/marcboeker/go-duckdb v1.0.8
 	github.com/mitchellh/mapstructure v1.5.0
+	github.com/senseyeio/duration v0.0.0-20180430131211-7c2a214ada46
```
[the remark this answers is not in the document]
Comment (pjain1) [reply go.mod]: copied

File: runtime/services/catalog/artifacts/yaml/objects.go [hunk 2]
```diff
@@ -159,6 +179,11 @@ func fromMetricsViewArtifact(metrics *MetricsView, path string) (*drivers.Catalo
 		measure.Name = fmt.Sprintf("measure_%d", i)
 	}
 
+	for _, timeGrain := range metrics.TimeGrains {
+		apiMetrics.TimeGrains = append(apiMetrics.TimeGrains, getTimeGrainEnum(timeGrain))
+	}
+	apiMetrics.DefaultTimeGrain = getTimeGrainEnum(metrics.DefaultTimeGrain)
```
[the remark this answers is not in the document]
Comment (pjain1) [reply runtime/services/catalog/artifacts/yaml/objects.go]: done

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**